prov/udp/src/udpx_ep.c

```diff
@@ -56,11 +56,25 @@ int udpx_setname(fid_t fid, void *addr, size_t addrlen)
 int udpx_getname(fid_t fid, void *addr, size_t *addrlen)
 {
 	struct udpx_ep *ep;
+	struct sockaddr_storage sock_addr = { 0 };
+	size_t sock_addr_len = sizeof(sock_addr);
 	int ret;
 
 	ep = container_of(fid, struct udpx_ep, util_ep.ep_fid.fid);
-	ret = getsockname(ep->sock, addr, (socklen_t *)addrlen);
-	return ret ? -errno : 0;
+	ret = getsockname(ep->sock, (struct sockaddr *)&sock_addr,
+			  (socklen_t *)&sock_addr_len);
+	if (ret)
+		return -ofi_sockerr();
+
+	if (*addrlen < sock_addr_len)
+		ret = -FI_ETOOSMALL;
```

Comment:
We can probably simplify the changes to:
```
size_t buflen = *addrlen;

ret = getsockname(ep->sock, addr, (socklen_t *) addrlen);
if (ret)
    return -errno;

return buflen < *addrlen ? -FI_ETOOSMALL : 0;
```

Reply:
Sure, thanks!

Reply:
I've updated the PR. But it doesn't work on Windows. Please, don't merge. I'll fix